core/domain/classifier_services.py

```diff
@@ -127,6 +127,69 @@ def classify_string_classifier_rule(state, normalized_answer):
     return None
 
 
+
+
+def add_to_training_queue(exploration):
```

Comment:
I agree that the calculation for states that don't need classifier mappings needs to be modified from my original suggestion, so the spirit of your suggestion sounds good to me. But I would advise against making up random dicts and passing them around; just pass two separate args for clarity.